### PR TITLE
Removes Tactical Resting

### PR DESCRIPTION
**Pull request by Fikou**
## About The Pull Request

If you are on combat mode, your bullets will hit people lying down or stunned, otherwise they pass through.

## Why It's Good For The Game

The original PR that added it was labeled a fix (#55495) only mentioning it in the changelog once, when it was a balance change, as resting to avoid bullets was intentionally removed after we added crawling and a rest button, for being fucking stupid.
The thing that original pr "fixed" was mobs being bullet sponges, which you can now prevent by switching off combat mode, so it is no longer needed.
Avoiding people shooting at you while also having guns of your own you are perfectly able to shoot is stupid.
It makes mobs like beepsky super annoying, where unless youre in a large area the robot chasing you with one of the only hardstuns left in the game will be really hard to hit.

## Changelog

:cl:
balance: If you are on Combat Mode, your bullets no longer pass through prone people
/:cl:

### CODE REVIEW

**Comment by CRITAWAKETS** [comment]
git gud lol

**Comment by Fikou** [comment]
> git gud lol

ive gotten gud with ur mom last night lmfao

**Comment by Fikou** [comment]
what the fuck its been 3 hours

**Comment by itseasytosee** [comment]
One day tactical resting will have its revenge.

**Comment by carshalash** [comment]
Does this only apply to bullets or is it fine for lasers too?

**Comment by Fikou** [comment]
> Does this only apply to bullets or is it fine for lasers too?

all guns and mech guns

**Comment by rellyreri** [comment]
ctf balance change

**Comment by Ryll-Ryll** [comment]
Honestly as kinda-cheesy of a strat as it is, it's also one of the only real techs we have in tg and comes with downsides of making you slower and easier to hit when targeted directly. I don't really care one way or the other balance wise, but it does make fights a bit more interesting to watch

**Comment by ATH1909** [comment]
god forbid hitting the deck to dodge a shot actually makes you harder to hit

also, I'm calling it now, using prone corpses as shields that you can shoot through but people in combat mode can't is gonna become a gamer trick for a while, since the ability of your shots to pass over corpses without hitting them being dependent on whether or not you're in combat mode is very unintuitive. how is someone supposed to know that if they don't keep their eyes glued to the changelogs or go code diving?

**Comment by ATH1909** [comment]
"Avoiding people shooting at you while also having guns of your own you are perfectly able to shoot is stupid."
just run up and baton them hello??

**Comment by Dawson1917** [comment]
As someone who's played on a downstream that removed tactical resting I can tell you its removal makes combat more boring. Because of the way projectiles work in ss13 it's always better to be aiming _behind_ someone and trying to shoot diagonally through tiles because it widens the hitbox. The only thing that hampered that meta was that somebody could lay down and now your shots are all missing because you weren't bothering to click anywhere near their sprite.

As long as people can't hotkey Rest so they have to stop shooting for a split second while they click the button, tactical resting gives a small amount of extra depth to gun combat.

**Comment by ATH1909** [comment]
> As long as people can't hotkey Rest so they have to stop shooting for a split second while they click the button, tactical resting gives a small amount of extra depth to gun combat.

We have a hotkey for resting, but it also takes a second to get up from resting.

**Comment by Livrah** [comment]
Idea: The game stops in the middle of the combat and we roll the dice to see who wins.

**Comment by Fikou** [comment]
> As long as people can't hotkey Rest so they have to stop shooting for a split second while they click the button, tactical resting gives a small amount of extra depth to gun combat.

but they can hotkey rest, we have a hotkey for resting


> We have a hotkey for resting, but it also takes a second to get up from resting.

pretty sure that do_after works even if you do stuff unlike most do_afters, so its not like that matters other than you are slightly slower for 1 second when getting up

> also, I'm calling it now, using prone corpses as shields that you can shoot through but people in combat mode can't is gonna become a gamer trick for a while

it wont i made sure :p

**Comment by Fikou** [comment]
> Idea: The game stops in the middle of the combat and we roll the dice to see who wins.

ok wait until april fools

**Comment by ATH1909** [comment]
> pretty sure that do_after works even if you do stuff unlike most do_afters, so its not like that matters other than you are slightly slower for 1 second when getting up

being prone is a pretty significant slowdown, though, and your opponent can fire off 1-2 shots at you while you're still trying to get up

**Comment by Fikou** [comment]
> > pretty sure that do_after works even if you do stuff unlike most do_afters, so its not like that matters other than you are slightly slower for 1 second when getting up
> 
> being prone is a pretty significant slowdown, though, and your opponent can fire 1-2 shots at you while you're still trying to get up

no they cant they cant hit you

**Comment by ATH1909** [comment]
> > > pretty sure that do_after works even if you do stuff unlike most do_afters, so its not like that matters other than you are slightly slower for 1 second when getting up
> > 
> > 
> > being prone is a pretty significant slowdown, though, and your opponent can fire 1-2 shots at you while you're still trying to get up
> 
> no they cant they cant hit you

just click on your very slow sprite??

**Comment by Pickle-Coding** [comment]
This is a terrible PR that will implement meat shields to defend against people in combat mode. Just click on their sprite, they're too slow to dodge while resting.

**Comment by itseasytosee** [comment]
My suggestion for a compromise would be to give you a second or two of bullet passover when you go prone. However I do like it in its current state, it feels thematic.

**Comment by Fikou** [comment]
> This is a terrible PR that will implement meat shields to defend against people in combat mode. Just click on their sprite, they're too slow to dodge while resting.

this never happened during the time before resting to avoid bullets was randomly reimplemented



> My suggestion for a compromise would be to give you a second or two of bullet passover when you go prone. However I do like it in its current state, it feels thematic.

i think youre mostly gonna rest around for like 1-3 seconds so this wouldnt accomplish much while needing to write special code for it

**Comment by Watermelon914** [comment]
Personally, I don't like this solution. Shooting people on the ground shouldn't be easier, people shooting others whilst on the ground should be harder.
This means you can't play dead anymore or drop to the ground to dodge incoming bullets from offscreen.

**Comment by Mickyan** [comment]
Huge mistake labeling this as balance so now we have to argue about this all over again

Original change: https://github.com/tgstation/tgstation/pull/40931 (literally a week after the addition of crawling because it was that cancerous when everyone figured it out)
Broken and fixed in 2019: https://github.com/tgstation/tgstation/pull/42241
Reappears in a fix PR undocumented: https://github.com/tgstation/tgstation/pull/55495

**Comment by Watermelon914** [comment]
I'll remove the DNM label if you make it so that stunned and incapacitated players do not get hit by this.

File: code/modules/projectiles/projectile.dm [hunk 1]
```diff
@@ -504,13 +504,13 @@
 			return TRUE
 		if(L.stat == DEAD)
 			return FALSE
-		if(!hit_prone_targets) // If target not able to use items, move and stand - or if they're just dead, pass over.
+		if(HAS_TRAIT(L, TRAIT_IMMOBILIZED) && HAS_TRAIT(L, TRAIT_FLOORED) && HAS_TRAIT(L, TRAIT_HANDS_BLOCKED))
```
Comment (Watermelon914) [inline code/modules/projectiles/projectile.dm:507]:
```suggestion
		if(AS_TRAIT(L, TRAIT_FLOORED) && (HAS_TRAIT(L, TRAIT_IMMOBILIZED)  || HAS_TRAIT(L, TRAIT_HANDS_BLOCKED)))
```

Comment (Fikou) [reply code/modules/projectiles/projectile.dm:507]: AS_TRAIT

Comment (Fikou) [reply code/modules/projectiles/projectile.dm:507]:
```suggestion
		if((HAS_TRAIT(L, TRAIT_IMMOBILIZED) && HAS_TRAIT(L, TRAIT_FLOORED)) || HAS_TRAIT(L, TRAIT_HANDS_BLOCKED))
```

File: code/modules/projectiles/projectile.dm [hunk 2]
```diff
@@ -504,15 +502,15 @@
 		var/mob/living/L = target
 		if(direct_target)
 			return TRUE
-		// If target not able to use items, move and stand - or if they're just dead, pass over.
 		if(L.stat == DEAD)
 			return FALSE
-		if(!L.density)
+		if((HAS_TRAIT(L, TRAIT_IMMOBILIZED) && HAS_TRAIT(L, TRAIT_FLOORED)) || HAS_TRAIT(L, TRAIT_HANDS_BLOCKED))
```
Comment (Fikou) [inline code/modules/projectiles/projectile.dm:507]:
```suggestion
		if(HAS_TRAIT(L, TRAIT_IMMOBILIZED) && HAS_TRAIT(L, TRAIT_FLOORED) && HAS_TRAIT(L, TRAIT_HANDS_BLOCKED))
```

**Comment by Livrah** [comment]
Why is "Why Good for a game" sounds like Why is this Bad for game. Add this for your CTF spawns and be happy.

**Comment by SinfulBliss** [comment]
This PR simplifies combat. Resting makes you incredibly slow in exchange for being more difficult to click with bullets - it's balanced. Also goes without saying this is a massive buff to guns like makarov and 357.

**Comment by CameronWoof** [comment]
> This is a terrible PR that will implement meat shields to defend against people in combat mode. Just click on their sprite, they're too slow to dodge while resting.

Projectiles are slow enough that given distance, it's functionally impossible to hit a resting person by clicking on them--they just writhe out of the way by the time the projectile travels there.

**Comment by DmitriyRubetskoy** [comment]
I have no idea why you'd see this as an actual issue.

Tactical resting is akin to diving for cover in combat, so it makes complete sense. Lying on the ground makes you slower, in exchange for being a harder target to hit.

**Comment by Fikou** [comment]
> I have no idea why you'd see this as an actual issue.
> 
> Tactical resting is akin to diving for cover in combat, so it makes complete sense. Lying on the ground makes you slower, in exchange for being a harder target to hit.

people diving for cover in combat dont crawl up to their target to shoot them, make some actual cover structures or use the ones scattered around the place to cover yourself

**Comment by itseasytosee** [comment]
You could distinguish crawling and laying down characters for the sake of bullet passover.

**Comment by ATH1909** [comment]
> > I have no idea why you'd see this as an actual issue.
> > Tactical resting is akin to diving for cover in combat, so it makes complete sense. Lying on the ground makes you slower, in exchange for being a harder target to hit.
> 
> people diving for cover in combat dont crawl up to their target to shoot them, make some actual cover structures or use the ones scattered around the place to cover yourself

so just add extra bullet spread for firing a gun while prone??

**Comment by Dorsisdwarf** [comment]
Personally I absolutely hate the fact that as soon as anyone opens fire everyone just hits the floor and starts jiggling about all over the place while shooting back. Never mind the absolute weirdness of people defending it because its "a tech" yeah like tactical combat chairs was "a tech" as was wavedashing when not dropping your stuff was. But we removed both of them because they sucked. "Here's my annoying gamey combat strategy that exploits the latest badly thought out patch but i call it a tech so you cant remove it"

This PR just creates new and stupider "techs" though so i guess its damned either way.

**Comment by optimumtact** [comment]
We nerfed this before for good reason, i'm not happy to see it still being used.

If you want to have tactical rest based combat, you can make a PR to that effect with a fully realised hackmd instead of trying to escalate a misfeature into a feature.

**Comment by KittyNoodle** [comment]
Literally 1983

**Comment by optimumtact** [comment]
deathbycow

**Comment by Rohesie** [comment]
The intent of this PR was good, but not its execution.
Before this PR and the prior one the behavior was that if you were lying down but weren't stunned, you'd still be hit.
That meant you couldn't drop into the ground and avoid shots, but dead bodies and stunned people wouldn't be blocking shots.

**Pull request merged**